File: examples/todo-java-spring/output/src/main/java/com/example/todo/TodoItemService.java
package com.example.todo;

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
@Service
public class TodoItemService {
    private final TodoItemRepository repository;
    public TodoItemService(TodoItemRepository repository) {
        this.repository = repository;
    }
    public java.util.List<TodoItem> findAll() {
        return repository.findAll();
    }
    public TodoItem findById(java.util.UUID id) {
        java.util.Optional<TodoItem> opt = repository.findById(id);
        if (!opt.isPresent()) {
            throw new java.util.NoSuchElementException("TodoItem not found: " + id);
        }
        return opt.get();
    }
    @Transactional
    public TodoItem create(TodoItem entity) {
        entity.setCreatedAt(java.time.LocalDateTime.now());
        return repository.save(entity);
    }
    @Transactional
    public TodoItem update(java.util.UUID id, TodoItem entity) {
        TodoItem existing = findById(id);
        existing.setName(entity.getName());
        existing.setDescription(entity.getDescription());
        existing.setUpdatedAt(java.time.LocalDateTime.now());
        return repository.save(existing);
    }
    @Transactional
    public void deleteById(java.util.UUID id) {
        repository.deleteById(id);
    }
}

File: examples/todo-kotlin-quarkus/output/src/main/kotlin/com/example/todo/TodoItemService.kt
package com.example.todo

import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import jakarta.transaction.Transactional
@ApplicationScoped
open class TodoItemService {
    @Inject
    lateinit var repository: TodoItemRepository
    fun findAll(): java.util.List<TodoItem> {
        return repository.listAll()
    }
    fun findById(id: Long): TodoItem? {
        return repository.findById(id)
    }
    @Transactional
    fun create(entity: TodoItem): TodoItem {
        entity.createdAt = java.time.LocalDateTime.now()
        repository.persist(entity)
        return entity
    }
    @Transactional
    fun update(id: Long, entity: TodoItem): TodoItem? {
        val existing = repository.findById(id) ?: return null
        existing.name = entity.name
        existing.description = entity.description
        existing.updatedAt = java.time.LocalDateTime.now()
        repository.persist(existing)
        return existing
    }
    @Transactional
    fun deleteById(id: Long): Boolean {
        return repository.deleteById(id)
    }
}